modules/utxo-lib/src/bitgo/Unspent.ts

```diff
@@ -97,11 +101,27 @@ export function toPrevOutput(u: Unspent, network: Network): PrevOutput {
  * @param u
  * @param sequence - sequenceId
  */
-export function addToTransactionBuilder(txb: UtxoTransactionBuilder, u: Unspent, sequence?: number): void {
+export function addToTransactionBuilder<TNumber extends number | bigint>(
+  txb: UtxoTransactionBuilder<TNumber>,
+  u: Unspent<TNumber>,
+  sequence?: number
+): void {
   const { txid, vout, script, value } = toPrevOutput(u, txb.network as Network);
   txb.addInput(txid, vout, sequence, script, value);
 }
 
-export function unspentSum(unspents: Unspent[]): number {
-  return unspents.reduce((sum, u) => sum + u.value, 0);
+export function unspentSum<TNumber extends number | bigint>(
+  unspents: Unspent<TNumber>[],
+  sumType: 'inherit' | 'number' | 'bigint' = 'inherit'
+): TNumber {
+  const total = unspents.reduce((sum, u) => sum + BigInt(u.value), BigInt(0));
+  if (sumType === 'bigint' || (unspents.length > 0 && typeof unspents[0].value === 'bigint' && sumType === 'inherit')) {
+    return total as TNumber;
+  } else {
```

Comment:
1. An empty unspents argument with `inherit` will cause the result `0` (`number`). I don't think that is what we want. We should error instead.
2. We are not checking if all unspents values have the same type.

Reply:
consider https://github.com/BitGo/BitGoJS/commit/165cbd92b11bace5dbe5ad0717a61649ea2b4dce

Reply:
I removed the 'inherit' option and defaulted to 'number', since an empty unspents argument would previously result in `0` (`number`) but that doesn't make sense anymore with the `inherit` sum type.

Reply:
I also used type assertions, and so if the unspents array mixes `bigint` and `number` there will be a runtime error

Reply:
I noticed that in https://github.com/BitGo/BitGoJS/commit/165cbd92b11bace5dbe5ad0717a61649ea2b4dce `toTNumber` was put in `src` code (instead of `test`) and exported - is that worth doing with `toTNumber` and `getValueScaled` (two helper functions I added to `transaction_util.ts`)

Reply:
I ended up moving `toTNumber` and `getValueScaled` to `src`